versioned_hdf5/tests/test_subchunk_map.py

```diff
@@ -1,136 +1,135 @@
-from collections import defaultdict
+from __future__ import annotations
+
+from typing import Any
 
 import hypothesis
-import ndindex
 import numpy as np
 import pytest
 from hypothesis import given
 from hypothesis import strategies as st
 from hypothesis.extra import numpy as stnp
-from numpy.testing import assert_equal
+from numpy.testing import assert_array_equal
```

Comment:
As I recall there is some difference here with `assert_equal`, in that `assert_equal` is defines special handling for non-array types and is thus a little more strict about what it considers equal. Are there any implications of changing to the more lenient `assert_array_equal`?

Reply:
From https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html:

> When either actual or desired is already an instance of [numpy.ndarray](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray) and desired is not a dict, the behavior of assert_equal(actual, desired) is identical to the behavior of assert_array_equal

The main difference is that in older versions of numpy assert_array_equal supports strict=True, while it's only been added later to assert_equal.